### PR TITLE
MM-47099 : Migrate "components/channel_notifications_modal/components/expand_view.jsx" to Typescript

### PR DESCRIPTION
**Pull request by VishakhaPoonia**
#### Summary
Migrate expand view to typescript

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/21070
Fixes https://mattermost.atlassian.net/browse/MM-47099

#### Related Pull Requests
NA

#### Screenshots
NA

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/channel_notifications_modal/components/expand_view.tsx [hunk 1]
```diff
@@ -15,6 +14,18 @@ import SettingItemMax from 'components/setting_item_max.jsx';
 import Describe from './describe.jsx';
 import ExtraInfo from './extra_info.jsx';
 import SectionTitle from './section_title.jsx';
+type Props = {
```
Comment (M-ZubairAhmed) [inline components/channel_notifications_modal/components/expand_view.tsx:17]: nit : extra line between

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: components/channel_notifications_modal/components/expand_view.tsx (hunk 1 above)
Comment (VishakhaPoonia) [reply components/channel_notifications_modal/components/expand_view.tsx:17]: Would love a rule for this in eslint.

Comment (M-ZubairAhmed) [reply components/channel_notifications_modal/components/expand_view.tsx:18]: yup we will add it. thank for suggesting

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @VishakhaPoonia 
Tested, looks good to merge.
Verified channel notifications modal is working as before.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/246202-1196d24-onprem-ent-webapp-pr-11219-mm-ee-test:1196d24/mochawesome.html)- no PR related failures.  
@M-ZubairAhmed Can you please approve the workflow and merge - thanks 🙂

**Comment by M-ZubairAhmed** [comment]
@VishakhaPoonia can you please resolve the conflict

**Comment by M-ZubairAhmed** [comment]
@VishakhaPoonia let us know when its ready we can merge it then

**Comment by VishakhaPoonia** [comment]
@M-ZubairAhmed it’s ready

**Comment by VishakhaPoonia** [comment]
@sinansonmez Could you please take another look?

**Pull request merged**